### PR TITLE
fix(nfc): Added retry attempt for card sign exception

### PR DESCRIPTION
**Pull request by amanCypherock**
https://app.clickup.com/t/9002019994/PRF-6294

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6294 On device card auth is failing on tapping 2nd card after 2nd beep](https://app.clickup.com/t/9002019994/PRF-6294)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6294 On device card auth is failing on tapping 2nd card after 2nd beep](https://app.clickup.com/t/9002019994/PRF-6294)

File: src/card_operations/card_sign.c [hunk 1]
```diff
@@ -126,10 +126,21 @@ static card_error_status_word_e get_card_auth_signature(uint8_t *sign_data,
...
+  uint16_t sign_data_size_inout = 0;
   card_error_status_word_e status = DEFAULT_UINT32_IN_FLASH_ENUM;
-  memcpy(sign_data_inout, sign_data, sign_data_size);
-  status = nfc_ecdsa(sign_data_inout, &sign_data_size_inout);
+
+  /**
+   * @brief This retry attempt has been added to avoid exceptions occuring due
+   * to incorrect signature length received from card. In a rare case the X1
+   * Card would send a incorrect signature with as the size would be one byte
+   * less than expected. When such a case is encountered, we do a retry attempt
```
Comment (ujjwal-cyph) [inline src/card_operations/card_sign.c:136]:
```suggestion
   * Card would send a incorrect signature with as the size would be less than
   * expected of 64-bytes. When such a case is encountered, we do a retry attempt
```

**Review by ujjwal-cyph — APPROVED** [review]
LGTM

**Pull request merged**